### PR TITLE
Fetch skill

### PR DESCRIPTION
**Pull request by psitosam**
_AP__ Wrote Tests
____ Implemented
____ Reviewed

## Necessary checkmarks:
- [x] All Tests are Passing
- [x] The code will run locally

## Type of change
- [x] New feature
- [] Bug Fix

## Implements/Fixes:
* Created tests and files to add endpoint for finding a skill by ID. 

## Check the correct boxes
- [x] This broke nothing
- [x] All Tests are Passing
- [x] The code will run locally

## Testing Changes
- [x] Tests have been added
- [] No Tests have been changed
- [] Some Tests have been changed
- [] All of the Tests have been changed(Please describe what in the world happened)

## Checklist:
- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code
- [x] I have removed any testing syntax (save_and_open_page/pry) in my spec files

### CODE REVIEW

File: spec/graphql/queries/fetch_skill_spec.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                expect(data[:criteria]).to be_a String
+
+                expect(data).to have_key(:youtubeLink)
+                expect(data[:youtubeLink]).to be_a String
+                
+            end 
+
+        end 
+
+        describe 'sad path' do
```
Comment (AliciaWatt) [inline spec/graphql/queries/fetch_skill_spec.rb:34]: nice work on the sad path!

File: app/graphql/mutations/create_user.rb [hunk 2]
```diff
@@ -2,15 +2,19 @@ class Mutations::CreateUser < Mutations::BaseMutation
...
-    if user = User.find_or_create_by!(username: username, email: email)
+    begin 
+    user = User.find_or_create_by(username: username, email: email)
       {user: user, errors: []}
-    else
-      {user: nil, errors: user.errors.full_messages}
+    
+    rescue ActiveRecord::ValidationFailed => _e
+    GraphQL::ExecutionError.new("Username has been taken.")
+    # else
```
Comment (AliciaWatt) [inline app/graphql/mutations/create_user.rb:16]: should we delete this commented-out code before the merge to main/

**Review by alexGrandolph — COMMENTED** [review]
really clean query

**Review by alexGrandolph — COMMENTED** [review]
thank you for taking care of this bud

**Comment by psitosam** [comment]
Hey y'all after seeing the local problems encountered by the team when trying to integrate my tests with subsequent tests, I'm going to try and go back to refactor the fetch dog spec to use interpolation instead of hard-coding the dog's id in the query.

**Pull request merged**